### ISSUE TITLE
make failed: No abi file found. Failed to download from github

### ISSUE THREAD

**Issue by 0xWilliamWang** [sigp/lighthouse#6346]
## Description
1. Hello everyone, I am new to lighthouse and rust. I am going to build this project from source code
2. I executed the `make` command in the container(rust:bookworm, https://hub.docker.com/_/rust/tags )and it failed.
3. It says that the file /ethereum/eth2.0-specs/v0.12.1/deposit_contract/contracts/validator_registration.json cannot be obtained. I guess it is trying to obtain the file from the eth2.0-specs repository of the ethereum organization, but this repository does not exist now. How can I fix this problem?   ` No abi file found. Failed to download from github: reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/ethereum/eth2.0-specs/v0.12.1/deposit_contract/contracts/validator_registration.json", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })) }
`
1. Any tips are much appreciated
## Version
 stable
```bash
root@5fde76e12718:/app# git log -1
commit 2e0eb6d1b8705bbda2ba56eb195d9cc7c6575e95 (HEAD -> stable, origin/stable, origin/HEAD)
Author: Age Manning <Age@AgeManning.com>
Date:   Thu Aug 29 14:44:34 2024 +1000

    Add retropgf funding (#6324)
root@5fde76e12718:/app# git status
On branch stable
Your branch is up to date with 'origin/stable'.

nothing to commit, working tree clean
```
## Present Behaviour
```bash
root@5fde76e12718:/app# make -e RUST_BACKTRACE=full     
cargo install --path lighthouse --force --locked \
        --features "" \
        --profile "release" \

  Installing lighthouse v5.3.0 (/app/lighthouse)
    Updating crates.io index
   Compiling deposit_contract v0.2.0 (/app/common/deposit_contract)
error: failed to run custom build command for `deposit_contract v0.2.0 (/app/common/deposit_contract)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `/app/target/release/build/deposit_contract-e5bdace01f716741/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at common/deposit_contract/build.rs:68:19:
  No abi file found. Failed to download from github: reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/ethereum/eth2.0-specs/v0.12.1/deposit_contract/contracts/validator_registration.json", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })) }
  stack backtrace:
     0:     0x561977f85e75 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hd736fd5964392270
     1:     0x561977faad1b - core::fmt::write::hc6043626647b98ea
     2:     0x561977f833cf - std::io::Write::write_fmt::h0d24b3e0473045db
     3:     0x561977f85c4e - std::sys_common::backtrace::print::h45eb8174d25a1e76
     4:     0x561977f86fe9 - std::panicking::default_hook::{{closure}}::haf3f0170eb4f3b53
     5:     0x561977f86d8a - std::panicking::default_hook::hb5d3b27aa9f6dcda
     6:     0x561977f87483 - std::panicking::rust_panic_with_hook::h6b49d59f86ee588c
     7:     0x561977f87364 - std::panicking::begin_panic_handler::{{closure}}::hd4c2f7ed79b82b70
     8:     0x561977f86339 - std::sys_common::backtrace::__rust_end_short_backtrace::h2946d6d32d7ea1ad
     9:     0x561977f87097 - rust_begin_unwind
    10:     0x561977825873 - core::panicking::panic_fmt::ha02418e5cd774672
    11:     0x56197782a6a4 - core::panicking::panic_display::h18cf9711d536f666
    12:     0x56197782ff96 - build_script_build::main::panic_cold_display::hff898934bbbe2883
    13:     0x56197782e4c8 - build_script_build::main::h7759544211f10f8f
    14:     0x5619778293b3 - core::ops::function::FnOnce::call_once::h309903ef30df14e6
    15:     0x5619778288e6 - std::sys_common::backtrace::__rust_begin_short_backtrace::he4d849203c340ee6
    16:     0x561977828da9 - std::rt::lang_start::{{closure}}::h9550e01c5ccb88ef
    17:     0x561977f7fd3d - std::rt::lang_start_internal::hcee5ed89fc25829a
    18:     0x561977828d87 - std::rt::lang_start::h1d4f768938e43701
    19:     0x56197782ffb5 - main
    20:     0x7f08698a724a - <unknown>
    21:     0x7f08698a7305 - __libc_start_main
    22:     0x561977826141 - _start
    23:                0x0 - <unknown>
error: failed to compile `lighthouse v5.3.0 (/app/lighthouse)`, intermediate artifacts can be found at `/app/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
make: *** [Makefile:43: install] Error 101
root@5fde76e12718:/app# 
```
## Expected Behaviour
Compilation succeeded

## Steps to resolve
1. clone repo
4. make

**Comment by dapplion** [comment]
If you are running the compilation dockerized this seems like an issues with your build setup / base image. Can you compile lighthouse on the host machine?

**Comment by michaelsproul** [comment]
Yeah even though the repo has been moved to `ethereum/consensus-specs` there is a redirect in effect.

The error you got makes it seem like Github is blocked or the connection is being redirected in a way that prevents it from resolving

**Comment by 0xWilliamWang** [comment]
@dapplion  Some problem

**Comment by 0xWilliamWang** [comment]
@michaelsproul @dapplion tks

**Comment by 0xWilliamWang** [comment]
For some reason, the router was resolving `raw.githubusercontent.com` to `127.0.0.1`; I solved this by setting `185.199.108.133 raw.githubusercontent.com` in `/etc/hosts`.
@michaelsproul @dapplion  Thank you very much for your tips
